### ISSUE TITLE
A lot of noise when logging in debug level

### ISSUE THREAD

**Issue by leonboot** [php-mqtt/client#165]
Hi, I noticed that, when running my Symfony application, which uses the php-mqtt/client library, with logging on debug level, the library generates a lot of noise in the logs. Each second, the following line appears around 10 times in the logs:

```json
{"message":"MQTT [{host}:{port}] [{clientId}] Read data from the socket (without blocking): {data}","context":{"host":"****","port":1883,"clientId":"****","data":""},"level":100,"level_name":"DEBUG","channel":"app","datetime":"2023-10-02T07:00:10.319721+00:00","extra":{}}
```

It seems the `readFromSocket` method always writes to the log, even if `$result`, the variable that contains the bytes read from the socket, contains an empty string: https://github.com/php-mqtt/client/blob/4d6ecec4d8ca21460d37258931b946a5fe7d28b2/src/MqttClient.php#L1232

Is this intentional or would this code benefit from being wrapped in an `if (!empty($result))` block?

**Comment by Namoshek** [comment]
Well, it really is meant to aid in debugging, so I think the log level is fine. You can, however, pass the `MqttClient` a different logger (with a lower log level) or filter the logs produced by the `MqttClient` via some kind of log filter.

If I was to change this, I would probably provide a separate configuration parameter for this behavior (as it is really useful in development and debugging of issues with the `MqttClient` and I don't want to miss it). However, this seems a bit difficult to integrate nicely at this point...

**Comment by leonboot** [comment]
> However, this seems a bit difficult to integrate nicely at this point...

Don't worry about it then. I only noticed it because by default, Monolog in my Symfony app is set to DEBUG level by default, so that same handler is passed through autowiring to the `MqttClient` class. It just seemed like a lot of noise, even for DEBUG levels, that read cycles that do not result in any message being read (`$result = ""`) also triggered a log entry. If it does actually serve a purpose (I suppose as a way of indicating the connection is still alive) that's a valid reason to leave it as it is.

I have been able to circumvent the issue in my project by adding an extra `mqtt` channel to Monolog and passing the logger for that channel to the `MqttClient` instance. I can now specify that only log entries with a level of INFO or higher should be logged.

Thanks for your quick response, I will close this ticket!
